src/providers/twitch/TwitchIrcServer.cpp

```diff
@@ -302,29 +302,62 @@ std::shared_ptr<Channel> TwitchIrcServer::getChannelOrEmptyByID(
     return Channel::getEmpty();
 }
 
+namespace {
+    // TODO: combine this with getEmoteSetBatches in TwitchAccount.cpp, maybe some templated thing
+    std::vector<QStringList> getChannelsInBatches(QStringList channels)
+    {
+        constexpr int batchSize = 100;
+
+        int batchCount = (channels.size() / batchSize) + 1;
+
+        std::vector<QStringList> batches;
+        batches.reserve(batchCount);
+
+        for (int i = 0; i < batchCount; i++)
+        {
+            QStringList batch;
+
+            // I hate you, msvc
+            int last = (std::min)(batchSize, channels.size() - batchSize * i);
+            for (int j = 0; j < last; j++)
+            {
+                batch.push_back(channels.at(j + (batchSize * i)));
+            }
+            batches.emplace_back(batch);
+        }
+
+        return batches;
+    }
+}  // namespace
+
 void TwitchIrcServer::bulkRefreshLiveStatus()
 {
-    QStringList userLogins;
-    this->forEachChannel([&userLogins](ChannelPtr chan) {
-        userLogins.push_back(chan->getName());
+    QStringList userIds;
```

Comment:
I know it's very inconsistent across the whole codebase, but when it comes to naming variables, Golang-like style is preferred for some elements like `ID`, `URL`, `API`, etc. We just don't want to rename every already existing variable without much gain, but newly created variables should follow that style.
```suggestion
    QStringList userIDs;
```
also make sure to update references to `userIds` below